Assets/_Functionalities/ObjectInformation/Scripts/BagInspector.cs

```diff
@@ -400,29 +396,29 @@ private void LoadFeatureContent(FeatureMapping mapping)
 		#endregion
 
 		#region UGUI methods
-		private void ShowObjectInformation()
+		public void ShowObjectInformation()
```

Comment:
Ik ben het met je eens om de selectors events te geven een hele mooie refactor slag is. Maar wat ik niet begrijp is om nu  deze methods via de inspector hier aan te koppelen met het oog op single responsibility wil je dit toch intern afhandelen

Reply:
Juist tegenovergesteld, door het met events op te lossen krijg je decoupling tussen het mechanisme van selectie en gaan tonen, en het daadwerkelijk renderen van de inhoud. Door deze losse coupling krijg je daarmee dat elk van die onderdelen een enkele individuele verantwoordelijkheid heeft en je kleinere classes en code krijgt. Nu kan ik een losse controller maken die ook extra functionaliteit toe gaat voegen zonder dat ik perse de BagInspector hoef aan te passen.

Reply:
in dat geval moeten we stellen dat deze 4 methods niet in deze class meer thuis horen

Reply:
Zie het als een Adapter pattern, we hebben een ObjectInspector die alleen als taak heeft om te weten wat er geselecteerd is (en zelfs dat zou ik los willen trekken zodat we dat kunnen hergebruiken) en wanneer die open en dicht moet, en verder delegeert die het tonen van specifieke informatie naar de adapters, die voor zichzelf kunnen besluiten of en welke data getoond gaat worden op basis van datgene wat geselecteerd is

Reply:
Ik begrijp je dnek ik niet helemaal maar volgens mij is de oplossing een visuele controller en een selectie controller voor de data. Zodat de methodes  in kwestie op de juiste plekken staan.  Decoupling met een event is dan volgnes mij een snelle oplossing om dit mogelijk te maken in dit geval. Maar omdat het een poc is laat ik hem voor nu